main.tf

```diff
@@ -73,3 +73,76 @@ module "cbr_rule" {
     tags = var.cbr_rules[count.index].tags
   }]
 }
+
+########################################################################################################################
+# SCC Workload Protection Trusted Profile
+########################################################################################################################
+
+# Create Trusted profile for SCC Workload Protection instance
+module "trusted_profile_scc_wp" {
+  count                       = var.cspm_enabled ? 1 : 0
+  source                      = "terraform-ibm-modules/trusted-profile/ibm"
+  version                     = "3.0.0"
+  trusted_profile_name        = var.scc_workload_protection_trusted_profile_name
+  trusted_profile_description = "Trusted Profile for SCC workload Protection instance ${ibm_resource_instance.scc_wp.guid} with required access for configuration aggregator."
+
+  trusted_profile_identity = {
+    identifier    = ibm_resource_instance.scc_wp.crn
+    identity_type = "crn"
+  }
+
+  trusted_profile_policies = [
+    {
+      unique_identifier = "scc-wp"
+      roles             = ["Service Configuration Reader", "Viewer", "Configuration Aggregator Reader"]
+      resources = [{
+        service = "apprapp"
+      }]
+      description = "App Config access"
+    },
+    {
+      unique_identifier = "scc-wp-enterprise"
+      roles             = ["Viewer", "Usage Report Viewer"]
+      resources = [{
+        service = "enterprise"
+      }]
+      description = "Enterprise access"
+    }
+  ]
+
+  trusted_profile_links = [{
+    unique_identifier = "scc-wp-vsi-link"
+    cr_type           = "VSI"
+    links = [{
+      crn = ibm_resource_instance.scc_wp.crn
+    }]
+  }]
+}
+
+################################################################
+# Enable CSPM for SCC Workload Protection instance
+################################################################
+
+# CSPM can only be enabled after the trusted profile exists,
+# but profile can only exist after instance has been created
+# hence we cannot directly enable CSPM in the instance creation
+# and need to use a separate resource to enable it
+resource "restapi_object" "enable_cspm" {
+  count = var.cspm_enabled ? 1 : 0
+
+  path = "/v2/resource_instances/${ibm_resource_instance.scc_wp.guid}"
+
+  data = jsonencode({
+    parameters = {
+      enable_cspm = true
+      target_accounts = [
+        {
+          account_id         = ibm_resource_instance.scc_wp.account_id
+          config_crn         = var.app_config_crn
+          trusted_profile_id = module.trusted_profile_scc_wp[0].profile_id
+        }
+      ]
+    }
+  })
+  create_method = "PATCH" # Specify the HTTP method for updates
```

Comment:
Can you add `force_new = var.cspm_enabled` - I think we will want this patch to run any time this boolean changes